### ISSUE TITLE
decorateAction not working as expected

### ISSUE THREAD

**Issue by shilman** [storybooks/storybook#776]
<a href="https://github.com/ritz078"><img src="https://avatars0.githubusercontent.com/u/5389035?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ritz078](https://github.com/ritz078)**
_Friday Oct 07, 2016 at 13:48 GMT_
_Originally opened as https://github.com/storybooks/storybook-addon-actions/issues/23_

----

So currently, if I do `args => args.splice(0, 1)` still I will get an array. How can I just log the first element of that array ?

if I do 

``` js
const firstArg = decorateAction([
  args => args[0]
]);
```

In simpler terms how do I log `firstTerm` instead of `[firstTerm]` if my `args` was `[firstTerm, secondTerm, thirdTerm]`

**Comment by shilman** [comment]
<a href="https://github.com/mnmtanish"><img src="https://avatars2.githubusercontent.com/u/136691?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mnmtanish](https://github.com/mnmtanish)**
_Monday Oct 10, 2016 at 09:23 GMT_

----

Hi @ritz078 

Action decorators taking an array of arguments as input and returning an array is intentional. It'll be easier to write action decorators this way.

For your case, try decorating the action manually

``` js
const firstArg = name => {
  const actionFn = action(name)
  return (...args) => actionFn(args[0])
}
```

**Comment by shilman** [comment]
Thanks for raising this issue. It seems like the issue's been answered so I'm going to close it for now. Please re-open if it's still an issue.
